File: app/src/main/java/com/platdmit/simplecloudmanager/app/vm/MainViewModel.java
package com.platdmit.simplecloudmanager.app.vm;

import com.platdmit.simplecloudmanager.domain.helpers.ActualApiKeyService;

import androidx.lifecycle.ViewModel;

public class MainViewModel extends ViewModel {
    private static final String TAG = LoginViewModel.class.getSimpleName();
    private ActualApiKeyService mActualApiKeyService;

    public MainViewModel(ActualApiKeyService actualApiKeyService) {
        mActualApiKeyService = actualApiKeyService;
    }

    @Override
    protected void onCleared() {
        super.onCleared();
        mActualApiKeyService.stopAutoUpdate();
    }
}

File: app/src/main/java/com/platdmit/simplecloudmanager/app/vm/MainViewModel.kt
package com.platdmit.simplecloudmanager.app.vm

import androidx.lifecycle.ViewModel
import com.platdmit.simplecloudmanager.domain.helpers.ActualApiKeyService

class MainViewModel(
        private val mActualApiKeyService: ActualApiKeyService
) : ViewModel() {
    companion object {
        private val TAG = LoginViewModel::class.java.simpleName
    }

    override fun onCleared() {
        super.onCleared()
        mActualApiKeyService.stopAutoUpdate()
    }
}